sdcm/nemesis.py

```diff
@@ -3537,7 +3538,38 @@ def disrupt_memory_stress(self):
         Try to run stress-ng to preempt allocated memory of scylla process,
         we don't monitor swap usage in /proc/$scylla_pid/status, just make sure
         no coredump, serious db error occur during the heavy load of memory.
+
+        For k8s cluster uses chaos-mesh experiment based on https://github.com/chaos-mesh/memStress
         """
+        if self._is_it_on_kubernetes():
+            if not self.tester.k8s_cluster.chaos_mesh.initialized:
+                raise UnsupportedNemesis(
+                    "Chaos Mesh is not installed. Set 'k8s_use_chaos_mesh' config option to 'true'")
+            self.log.info('Try to allocate 120% available memory, the allocated memory will be swaped out')
+            memory_limit = self.tester.k8s_cluster.calculated_memory_limit
+            memory, suffix = int(memory_limit[:-2]), memory_limit[-2:]
```

Comment:
At first, need to use existing utilities.
For this case we have `convert_memory_value_from_k8s_to_units` function.

At second, I think such conversion must be part of the `MemoryStressExperiment` class itself.
Consumer must not be bothered with such details.
So, consumer should just specified the multiplier.
Example:
```
experiment = MemoryStressExperiment(
    pod=self.target_node,
    duration="160s",
    workers=4,
    memory_multiplier=1.2,
    time_to_reach="60s",
)
```

Reply:
MemoryStressExperiment supports `%`: we can provide `90%` or in absolute units, so I'll stay with `%` support.
The reason I do the tricks here with memory limits read is due not supporting values >100% by chaos-mesh. 

I tried to keep the test as much as possible to VM based test. Although, I'm not sure if it is relevant, as from what I've read, K8s is rather not using swappiness (from the other hand, when reading `/sys/fs/cgroup/memory/memory.swappiness` it returns 10), is it?. And another question - will k8s allow memory to exceed the limit?
Maybe we can adapt the test and use 100% instead 120% for k8s?

Reply:
the way we configured the pods with limit/request identical, k8s doesn't let it use more memory then available.

swap is k8s is a recently new addition (only introduce in 1.22): 
https://kubernetes.io/blog/2021/08/09/run-nodes-with-swap-alpha/

I think doing 100% for now it enough, I guess later when swap is a bit more commonly used in k8s, chaos mesh would support higher then 100%, or some other configuration values to related to swap